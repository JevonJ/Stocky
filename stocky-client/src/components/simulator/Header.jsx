import React from 'react';

const Header = () => (
  <nav class="navbar navbar-dark sticky-top bg-dark flex-md-nowrap p-0">
    <a class="navbar-brand col-sm-3 col-md-2 mr-0" href="#">Stocky</a>
    <ul class="navbar-nav px-3">
      <li class="nav-item text-nowrap">
        <a class="nav-link" href="#">Sign out</a>
      </li>
    </ul>
  </nav>
);

export default Header;
