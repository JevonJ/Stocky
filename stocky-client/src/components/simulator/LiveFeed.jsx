import React, { Component } from 'react';
import { Card, CardTitle, CardText } from 'reactstrap';

class LiveFeed extends Component {
  static renderELemets({
    username, symbol, unitPrice, qauntity, type,
  }) {

    return (<CardText>{`${username} ${type} ${qauntity} from ${symbol} at ${unitPrice}`}</CardText>);
  }

  render() {
    const { liveFeed } = this.props;
    return (
      <Card body outline color="primary">
        <CardTitle>Live feed</CardTitle>
        {
          liveFeed.map(feedelement => LiveFeed.renderELemets(feedelement))
        }
      </Card>
    );
  }
}

export default LiveFeed;
