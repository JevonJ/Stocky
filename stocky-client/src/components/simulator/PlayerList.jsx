import React, { Component } from 'react';
import { Card, CardText, CardTitle, Table } from 'reactstrap';

class PlayerList extends Component {
  static renderPlayers(player, index) {
    return (
      <tr>
        <td>{index + 1}</td>
        <td>{player.name}</td>
        <td>LKR {player.cash}</td>
      </tr>
    );
  }

  render() {
    const { players } = this.props;

    return (
      <Card body outline color="danger">
        <CardTitle><h5>Room name: Team Titans</h5></CardTitle>
        <CardText>
          <Table borderless>
            <thead>
              <tr>
                <th>Rank </th>
                <th>Name</th>
                <th>Assets</th>
              </tr>
            </thead>
            <tbody>
              {
                players.map((player, index) => PlayerList.renderPlayers(player, index))
              }
            </tbody>
          </Table>
        </CardText>
      </Card>
    );
  }
}

export default PlayerList;
