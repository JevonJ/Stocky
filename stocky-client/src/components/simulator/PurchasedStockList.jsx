import React, { Component } from 'react';
import { Table, Collapse, Button, Row } from 'reactstrap';

import SellShareModal from '../modals/SellSharesModal';

class PurchasedStockList extends Component {
  constructor(props) {
    super(props);
    this.state = {
      sellModal: false,
    };
  }
  toggleSellModal(sellStockData) {
    const { toggleSellModal } = this.props;
    toggleSellModal();
    this.setState({
      selectedSellStock: sellStockData,
    });
  }

  renderPurchasedStocks(purchasedStocks, index) {
    const { roomStocks } = this.props;
    const stockPriceArr = roomStocks[purchasedStocks.stockSymbol];
    const stockPrice = stockPriceArr[stockPriceArr.length - 1];
    const totalSoldQty = purchasedStocks.soldStockQty.reduce((a, b) => a + b, 0);
    const remainingStocks = purchasedStocks.initStockQty - totalSoldQty;

    if (remainingStocks > 0) {
      return (
        <tr key={purchasedStocks.stockSymbol}>
        <td>{index + 1}</td>
        <td>{purchasedStocks.stockSymbol}</td>
        <td>{remainingStocks}</td>
        <td>{(purchasedStocks.unitPrice).toFixed(2)}</td>
        <td>{stockPrice.toFixed(2)}</td>

        <td>
          <Button
            size="sm"
            color="danger"
            onClick={() => this.toggleSellModal({
            symbol: purchasedStocks.stockSymbol,
            oldPrice: purchasedStocks.unitPrice,
            curPrice: stockPrice,
            avlqty: purchasedStocks.initStockQty,
            })}
          >
            Sell
          </Button>
        </td>
      </tr>
      );
    }
  }

  render() {
    const {
      socket, playerStocks, user, isOpen, sellModalState, roomInfo,
    } = this.props;

    return (
      <Row>
        <SellShareModal
          isOpen={sellModalState}
          toggle={() => this.toggleSellModal()}
          sellStockData={this.state.selectedSellStock}
          socket={socket}
          user={user}
          playerStocks={playerStocks}
          roomInfo={roomInfo}
        />
        <Collapse isOpen={isOpen}>
          <Table striped responsive size="sm">
            <thead>
              <tr>
                <th>#</th>
                <th>Symbol</th>
                <th>Qty</th>
                <th>Value</th>
                <th>Current</th>
              </tr>
            </thead>
            <tbody>
              {
                playerStocks[user.name] &&
                playerStocks[user.name].purchased
                .map((purchasedStocks, index) => this.renderPurchasedStocks(purchasedStocks, index))
              }
            </tbody>
          </Table>
        </Collapse>
      </Row>
    );
  }
}

export default PurchasedStockList;
