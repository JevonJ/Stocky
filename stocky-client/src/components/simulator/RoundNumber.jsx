import React, { Component } from 'react';
import { Card, CardTitle, CardText } from 'reactstrap';

class RoundNumber extends Component {
  render() {
    const { roomInfo, user } = this.props;
    return (
      <Card body outline color="primary">
        <CardTitle>RoundNumber</CardTitle>
        <CardText>{roomInfo[user.room].currentRound} / {roomInfo[user.room].rounds}</CardText>
      </Card>
    );
  }
}

export default RoundNumber;
