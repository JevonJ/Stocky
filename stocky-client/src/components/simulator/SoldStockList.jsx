import React, { Component } from 'react';
import { Table, Collapse, Row } from 'reactstrap';

class SoldStockList extends Component {
  static renderSoldStocks(soldStocks, index) {
    const totalQty = soldStocks.stockQty.reduce((a, b) => a + b, 0);
    return (
      <tr key={soldStocks.stockSymbol}>
        <td>{index + 1}</td>
        <td style={{ textAlign: 'center' }}>{soldStocks.round}</td>
        <td>{soldStocks.stockSymbol}</td>
        <td>{totalQty}</td>
        <td>{(soldStocks.unitPrice).toFixed(2)}</td>
      </tr>
    );
  }


  render() {
    const { playerStocks, user, isOpen } = this.props;

    return (

      <Row>
        <Collapse isOpen={isOpen}>
          <Table striped responsive size="sm">
            <thead>
              <tr>
                <th>#</th>
                <th>Round</th>
                <th>Symbol</th>
                <th>Qty</th>
                <th>Value</th>
              </tr>
            </thead>
            <tbody>
              {
                 playerStocks[user.name] &&
                 playerStocks[user.name].sold
                 .map((soldStocks, index) => SoldStockList.renderSoldStocks(soldStocks, index))
              }
            </tbody>
          </Table>
        </Collapse>
      </Row>
    );
  }
}

export default SoldStockList;
