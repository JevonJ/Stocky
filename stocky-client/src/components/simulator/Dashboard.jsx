import React, { Component } from 'react';
import { connect } from 'react-redux';
import { Row, Col, ListGroup, ListGroupItem, ButtonGroup, Button, Card, CardTitle, ButtonDropdown, DropdownToggle, DropdownMenu, DropdownItem, Input, Label, CardBody } from 'reactstrap';
import CountDown from 'react-countdown-clock';
import ReactLoading from 'react-loading';

import BuyModal from '../modals/BuyModalMain';
import DashboardHeaderData from './DashboarHeaderData';
import PlayerList from './PlayerList';
import LiveFeed from './LiveFeed';
import StockList from './StockList';
import SoldStockList from './SoldStockList';
import PurchasedStockList from './PurchasedStockList';
import RoundNumber from './RoundNumber';

import { setTime } from '../../actions';

const styles = {
  overlay: {
    width: '100%',
    height: '100%',
    position: 'absolute',
    top: 0,
    left: 0,
    backgroundColor: 'rgba(33,33,33 ,0.9)',
    zIndex: 99,
    paddingTop: '35vh',
  },
};

class Dashboard extends Component {
  constructor(props) {
    super(props);
    this.state = {
      collapse: false,
      collapse1: false,
      news: [],
      modal: false,
      sellModal: false,
    };
  }

  componentWillMount() {
    const { user, history } = this.props;
    if (user.constructor === Object && Object.keys(user).length === 0) {
      history.replace('/login');
    }
  }

  toggle(type) {
    if (type === 'collapse') this.setState({ collapse: !this.state.collapse });
    if (type === 'collapse1') this.setState({ collapse1: !this.state.collapse1 });
  }

  addTonews() {
    const newArr = this.state.news;
    newArr.push('newsssssss');
    this.setState({
      news: newArr,
    });
  }

  toggleModal(stock) {
    this.setState({
      modal: !this.state.modal,
      selectedStock: stock,
    });
  }

  toggleSellModal() {
    this.setState({
      sellModal: !this.state.sellModal,
    });
  }

  sortBySector() {
    this.setState({

    });
  }

  calculateStocks() {
    const { user, socket } = this.props;

    this.setState({
      sellModal: false,
      modal: false,
    });

    this.props.setTime({ round_time: 0 });

    if (user.host) {
      socket.emit('calculate_stocks', user.room);
    }
  }

  render() {
    const {
      socket, players, playerStocks, stocks, sectors, sectorStocks, stockInfo, liveFeed, roomStocks, user, roomInfo, time
    } = this.props;
    return (
      <Row>
        {
          (time.start_time > 0) &&
          <div id="Overlay" style={styles.overlay}>
            <Col
              sm="12"
              style={{
                color: '#8c98a5', fontWeight: 'bold', fontSize: '2em', textAlign: 'center', padding: '1em',
              }}
            >
              Game starting in...
            </Col>
            <div style={{ paddingLeft: '45%' }}>
              <CountDown
                seconds={time.start_time}
                color="#fff"
                alpha={0.9}
                size={100}
                onComplete={() => { this.props.setTime({ start_time: 0 }); socket.emit('start_game', user.room); }}
              />
            </div>
          </div>
        }

        {time.round_time === 0 && time.start_time === 0 &&
          <div id="Overlay" style={styles.overlay}>
            <Col
              sm="12"
              style={{
                color: '#8c98a5', fontWeight: 'bold', fontSize: '2em', textAlign: 'center',
              }}
            >
              Loading next round.
            </Col>
            <div style={{ paddingLeft: '46%' }}>
              <ReactLoading type="bars" color="#fff" width={100} height={100} />
            </div>
          </div>
        }

        <BuyModal
          isOpen={this.state.modal}
          toggle={() => this.toggleModal()}
          stock={this.state.selectedStock}
          roomStocks={roomStocks}
          roomInfo={roomInfo}
          socket={socket}
        />

        <Col xs="3">
          <Row>
            <Button outline color="primary" onClick={() => this.toggle('collapse')} style={{ marginBottom: '1rem' }}><h4>Sold Stocks </h4></Button>
          </Row>
          <SoldStockList
            isOpen={this.state.collapse}
            playerStocks={playerStocks}
            user={user}
          />
          <Row>
            <Button outline color="primary" onClick={() => this.toggle('collapse1')} style={{ marginBottom: '1rem' }}><h4>Purchased </h4></Button>
          </Row>
          <PurchasedStockList
            isOpen={this.state.collapse1}
            socket={socket}
            playerStocks={playerStocks}
            roomStocks={roomStocks}
            user={user}
            toggleSellModal={() => this.toggleSellModal()}
            sellModalState={this.state.sellModal}
          />
          <Row>
            <Card
              body
              inverse
              style={{
                  backgroundColor: '#333', borderColor: '#333', padding: '0.5rem', alignItems: 'center',
                }}
            >
              <CardTitle>Symbol lookup</CardTitle>
              <CardBody>
                <ButtonDropdown
                  isOpen={this.state.btnDropright}
                  toggle={() => this.setState({ btnDropright: !this.state.btnDropright })}
                >
                  <DropdownToggle outline caret>
                      Search by
                  </DropdownToggle>
                  <DropdownMenu>
                    <DropdownItem>Company name</DropdownItem>
                    <DropdownItem>Symbol</DropdownItem>
                  </DropdownMenu>
                </ButtonDropdown>
                <Input placeholder="" style={{ marginTop: '1rem', marginBottom: '1rem' }} />
                <Label style={{ textAlign: 'center' }}>Arpico(Pvt)Ltd</Label>
              </CardBody>
            </Card>
          </Row>
        </Col>
        <Col xs="6">
          <DashboardHeaderData user={user} playerStocks={playerStocks} />
          <Row>
            <h2>Currently in Market {'>>>'}</h2>
          </Row>
          <Row>
            <ButtonGroup>
              <Button outline color="info">Sort by sector</Button>
              <Button outline color="info">Sort by price</Button>
            </ButtonGroup>
          </Row>
          <Row>
            {Object.keys(user).length !== 0 &&
            <StockList
              sectors={sectors}
              sectorStocks={sectorStocks}
              stocks={stocks}
              stockInfo={stockInfo}
              roomStocks={roomStocks}
              toggleModal={stock => this.toggleModal(stock)}
            />
              }
          </Row>
        </Col>
        <Col xs="3">
          <Row>
            <Col sm="6">
              <Row>
                {roomInfo[user.room] && <RoundNumber roomInfo={roomInfo} user={user} />}
              </Row>
            </Col>
            <Col sm="6">
              <Row>
                <p><b>Time is Running!!!</b></p>
              </Row>
              <Row>
                {time.round_time && time.round_time > 0 &&
                  <CountDown
                    seconds={time.round_time}
                    color="#000"
                    alpha={0.9}
                    size={100}
                    onComplete={() => this.calculateStocks()}
                  />
                }
              </Row>
            </Col>
          </Row>
          <Row>
            <PlayerList players={players} playerStocks={playerStocks} />
          </Row>
          <Row>
            <LiveFeed liveFeed={liveFeed} />
          </Row>
          <Row>
            <Card body outline color="warning">
              <CardTitle>Current Events</CardTitle>
              <CardBody>
                <ListGroup>
                  <ListGroupItem>Engineering sector is having a boom of/... </ListGroupItem>
                  <ListGroupItem>ETI Group is undergoing a scandal...</ListGroupItem>
                </ListGroup>
              </CardBody>
              <Button outline color="secondary">Clear</Button>
            </Card>
          </Row>
        </Col>
      </Row>
    );
  }
}

const mapStateToProps = ({
  stocks, sectors, sectorStocks, liveFeed, players, playerStocks, stockInfo, roomStocks, user, roomInfo, time,
}) => ({
  players,
  stocks,
  sectors,
  playerStocks,
  sectorStocks,
  roomInfo,
  stockInfo,
  liveFeed,
  roomStocks,
  user,
  time,
});

export default connect(mapStateToProps, { setTime })(Dashboard);

