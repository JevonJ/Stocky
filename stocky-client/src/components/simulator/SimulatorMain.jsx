import React, { Component } from 'react';
import { connect } from 'react-redux';
import { Button, Fade, Popover, PopoverBody, PopoverHeader } from 'reactstrap';
import './SimulatorMain.css';

//import Sidebar from './Sidebar';
import Header from './Header';
import Dashboard from './Dashboard';

class SimulatorMain extends Component {
  render() {
    return (
      <div>
        <Header />
        <div class="container-fluid">
          <div class="p-3 mb-2 bg-gradient-dark text-black">
            <Dashboard />
          </div>
        </div>
      </div>
    );
  }
}

const mapStateToProps = (state) => {
  console.log('redux state:', state);
};

export default connect(mapStateToProps)(SimulatorMain);
