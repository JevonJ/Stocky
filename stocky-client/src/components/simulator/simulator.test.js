import React from 'react';
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';

import DashboardHeaderData from './DashboarHeaderData';
import Header from './Header';
import StockList from './StockList';
import SoldStockList from './SoldStockList';
import SimulatorMain from './SimulatorMain';
import PlayerList from './PlayerList';
import LiveFeed from './LiveFeed';
import RoundNumber from './RoundNumber';
import PurchasedStockList from './PurchasedStockList';
import Dashboard from './Dashboard';

const middlewares = [thunk];
const mockStore = configureStore(middlewares)
const initialState = { rooms: [], user: {}, players: [] };
const store = mockStore(initialState);

it('DashboardHeaderData renders correctly', () => {
  const wrapper = shallow(<DashboardHeaderData user={{ name: 'jevon' }} playerStocks={{ jevon: { purchased: [], sold: [] } }}/>);

  expect(wrapper).toMatchSnapshot();
});

it('Header renders correctly', () => {
  const wrapper = shallow(<Header />);

  expect(wrapper).toMatchSnapshot();
});

it('SimulatorMain renders correctly', () => {
  const wrapper = shallow(<SimulatorMain />);

  expect(wrapper).toMatchSnapshot();
});

it('Header renders correctly', () => {
  const wrapper = shallow(<Header />);

  expect(wrapper).toMatchSnapshot();
});

it('StockList renders correctly', () => {
  const wrapper = shallow(<StockList />);

  expect(wrapper).toMatchSnapshot();
});

it('SoldStockList renders correctly', () => {
  const wrapper = shallow(<SoldStockList playerStocks={{ Inuri: { sold: [] } }} user={{ name: 'Inuri' }} />);

  expect(wrapper).toMatchSnapshot();
});

it('PurchasedStockList renders correctly', () => {
  const wrapper = shallow(<PurchasedStockList playerStocks={{ Inuri: { purchased: [] } }} user={{ name: 'Inuri' }} />);

  expect(wrapper).toMatchSnapshot();
});

it('SimulatorMain renders correctly', () => {
  const wrapper = shallow(<SimulatorMain />);

  expect(wrapper).toMatchSnapshot();
});

it('PlayerList renders correctly', () => {
  const wrapper = shallow(<PlayerList players={[]} />);

  expect(wrapper).toMatchSnapshot();
});

it('LiveFeed renders correctly', () => {
  const wrapper = shallow(<liveFeed LiveFeed={[]} />);

  expect(wrapper).toMatchSnapshot();
});

it('RoundNumber renders correctly', () => {
  const wrapper = shallow(<RoundNumber user={{ room: 'Room1' }} roomInfo={{ Room1: { currentRound: 1 } }} />);

  expect(wrapper).toMatchSnapshot();
});

it('Dashboard renders correctly', () => {
  const wrapper = shallow(<Dashboard store={store} />);

  expect(wrapper).toMatchSnapshot();
});