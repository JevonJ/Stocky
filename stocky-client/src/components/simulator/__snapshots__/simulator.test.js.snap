// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard renders correctly 1`] = `
<Dashboard
  players={Array []}
  setTime={[Function]}
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
  storeSubscription={
    Subscription {
      "listeners": Object {
        "clear": [Function],
        "get": [Function],
        "notify": [Function],
        "subscribe": [Function],
      },
      "onStateChange": [Function],
      "parentSub": undefined,
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "unsubscribe": [Function],
    }
  }
  user={Object {}}
/>
`;

exports[`DashboardHeaderData renders correctly 1`] = `
<Row
  tag="div"
>
  <Col
    tag="div"
    widths={
      Array [
        "xs",
        "sm",
        "md",
        "lg",
        "xl",
      ]
    }
  >
    <ListGroup
      tag="ul"
    >
      <ListGroupItem
        className="justify-content-between"
        color="danger"
        tag="li"
      >
        <h4>
          Cash Remaining
        </h4>
        <h5>
          <Badge
            color="secondary"
            pill={true}
            tag="span"
          >
            LKR 
          </Badge>
        </h5>
      </ListGroupItem>
    </ListGroup>
  </Col>
  <Col
    tag="div"
    widths={
      Array [
        "xs",
        "sm",
        "md",
        "lg",
        "xl",
      ]
    }
  >
    <ListGroup
      tag="ul"
    >
      <ListGroupItem
        className="justify-content-between"
        color="warning"
        tag="li"
      >
        <h4>
          Stock Value
        </h4>
        <h5>
          <Badge
            color="secondary"
            pill={true}
            tag="span"
          >
            LKR 
            0
          </Badge>
        </h5>
      </ListGroupItem>
    </ListGroup>
  </Col>
  <Col
    tag="div"
    widths={
      Array [
        "xs",
        "sm",
        "md",
        "lg",
        "xl",
      ]
    }
  >
    <ListGroup
      tag="ul"
    >
      <ListGroupItem
        className="justify-content-between"
        color="info"
        tag="li"
      >
        <h4>
          Total Asset Value
        </h4>
        <h5>
          <Badge
            color="secondary"
            pill={true}
            tag="span"
          >
            LKR 
            NaN
          </Badge>
        </h5>
      </ListGroupItem>
    </ListGroup>
  </Col>
</Row>
`;

exports[`Header renders correctly 1`] = `
<nav
  className="navbar navbar-dark sticky-top bg-dark flex-md-nowrap p-0"
>
  <a
    className="navbar-brand col-sm-3 col-md-2 mr-0"
    href="#"
  >
    Stocky
  </a>
  <ul
    className="navbar-nav px-3"
  >
    <li
      className="nav-item text-nowrap"
    >
      <a
        className="nav-link"
        href="#"
      >
        Sign out
      </a>
    </li>
  </ul>
</nav>
`;

exports[`Header renders correctly 2`] = `
<nav
  className="navbar navbar-dark sticky-top bg-dark flex-md-nowrap p-0"
>
  <a
    className="navbar-brand col-sm-3 col-md-2 mr-0"
    href="#"
  >
    Stocky
  </a>
  <ul
    className="navbar-nav px-3"
  >
    <li
      className="nav-item text-nowrap"
    >
      <a
        className="nav-link"
        href="#"
      >
        Sign out
      </a>
    </li>
  </ul>
</nav>
`;

exports[`LiveFeed renders correctly 1`] = `
<liveFeed
  LiveFeed={Array []}
/>
`;

exports[`PlayerList renders correctly 1`] = `
<Card
  body={true}
  color="danger"
  outline={true}
  tag="div"
>
  <CardTitle
    tag="h5"
  >
    Room name: Team Titans
  </CardTitle>
  <CardBody
    tag="div"
  >
    <Table
      borderless={true}
      responsiveTag="div"
      tag="table"
    >
      <thead>
        <tr>
          <th>
            Rank 
          </th>
          <th>
            Name
          </th>
          <th>
            Assets
          </th>
        </tr>
      </thead>
      <tbody />
    </Table>
  </CardBody>
</Card>
`;

exports[`PurchasedStockList renders correctly 1`] = `
<Row
  tag="div"
>
  <SellSharesModal
    playerStocks={
      Object {
        "Inuri": Object {
          "purchased": Array [],
        },
      }
    }
    toggle={[Function]}
    user={
      Object {
        "name": "Inuri",
      }
    }
  />
  <Collapse
    appear={false}
    enter={true}
    exit={true}
    in={false}
    isOpen={false}
    mountOnEnter={false}
    onEnter={[Function]}
    onEntered={[Function]}
    onEntering={[Function]}
    onExit={[Function]}
    onExited={[Function]}
    onExiting={[Function]}
    tag="div"
    timeout={350}
    unmountOnExit={false}
  >
    <Table
      responsive={true}
      responsiveTag="div"
      size="sm"
      striped={true}
      tag="table"
    >
      <thead>
        <tr>
          <th>
            #
          </th>
          <th>
            Symbol
          </th>
          <th>
            Qty
          </th>
          <th>
            Value
          </th>
          <th>
            Current
          </th>
        </tr>
      </thead>
      <tbody />
    </Table>
  </Collapse>
</Row>
`;

exports[`RoundNumber renders correctly 1`] = `
<Card
  body={true}
  color="primary"
  outline={true}
  tag="div"
>
  <CardTitle
    tag="h5"
  >
    RoundNumber
  </CardTitle>
  <CardText
    tag="p"
  >
    1
     / 
  </CardText>
</Card>
`;

exports[`SimulatorMain renders correctly 1`] = `
<Container
  fluid={true}
  tag="div"
>
  <Connect(Dashboard) />
</Container>
`;

exports[`SimulatorMain renders correctly 2`] = `
<Container
  fluid={true}
  tag="div"
>
  <Connect(Dashboard) />
</Container>
`;

exports[`SoldStockList renders correctly 1`] = `
<Row
  tag="div"
>
  <Collapse
    appear={false}
    enter={true}
    exit={true}
    in={false}
    isOpen={false}
    mountOnEnter={false}
    onEnter={[Function]}
    onEntered={[Function]}
    onEntering={[Function]}
    onExit={[Function]}
    onExited={[Function]}
    onExiting={[Function]}
    tag="div"
    timeout={350}
    unmountOnExit={false}
  >
    <Table
      responsive={true}
      responsiveTag="div"
      size="sm"
      striped={true}
      tag="table"
    >
      <thead>
        <tr>
          <th>
            #
          </th>
          <th>
            Round
          </th>
          <th>
            Symbol
          </th>
          <th>
            Qty
          </th>
          <th>
            Value
          </th>
        </tr>
      </thead>
      <tbody />
    </Table>
  </Collapse>
</Row>
`;

exports[`StockList renders correctly 1`] = `
<Table
  responsive={true}
  responsiveTag="div"
  size="xs"
  striped={true}
  tag="table"
>
  <thead>
    <tr>
      <th>
        Company name
      </th>
      <th>
        Symbol
      </th>
      <th>
        Sector
      </th>
      <th>
        Last
      </th>
      <th>
        Current
      </th>
      <th />
    </tr>
  </thead>
  <tbody />
</Table>
`;
